flashinfer/grouped_mm/core.py

```
@@ -576,3 +578,263 @@ def grouped_mm_fp4(
+    """
+    if scale_major_mode != "MN":
+        raise ValueError(
+            f'scale_major_mode must be "MN" (kernel currently only supports the '
+            f'per-token MN-major TMA-aligned UE8M0 scale layout); got "{scale_major_mode}"'
+        )
+
+
+@supported_compute_capability([120, 121])
+@flashinfer_api
+def quantize_mxfp8_for_zero_padding(
+    input: torch.Tensor,
```

Ruff (0.15.15)

[error] Function argument `input` is shadowing a Python builtin

(A002)